src/app/styles/logo/logo-demo.component.ts

```diff
@@ -1,8 +1,28 @@
-import {Component} from '@angular/core';
+import { AfterViewInit, Component } from '@angular/core';
+import { Router } from '@angular/router';
 
 @Component({
     selector: 'hc-logo',
     templateUrl: './logo-demo.component.html',
     styleUrls: ['./logo-demo.component.scss']
 })
-export class LogoDemoComponent {}
+export class LogoDemoComponent implements AfterViewInit {
+    private section: string | null;
+
+    constructor(private router: Router) {}
+
+    ngAfterViewInit() {
+        this.section = this.extractUrlValue( 'section', this.router.url );
```

Comment:
This function exists everywhere... couldn't we DRY it up by extracting it out?

Reply:
Perhaps a small SectionScrollService?  The service could listen to the Router's URL change to store the current `section`, then the component calls `scrollToSection()` in `ngAfterViewInit`?